### PR TITLE
Fix rendering of const keyword for functions

### LINKED ISSUE #44187: rustdoc doesn't render constness of bare functions on nightly
On today's nightly docs for `std::mem::size_of`:

image

But if you click `[src]`, you can see that it's supposed to be const:

image

Rustdoc is supposed to print the constness of functions on nightly. It currently works for associated functions (e.g. `AtomicBool::new`) but bare functions like `size_of` are currently not working.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #44187.

r? @QuietMisdreavus

### CODE REVIEW

**Comment by QuietMisdreavus** [comment]
So there's something weird i noticed when digging into this. I saw that you took out the import for `UnstableFeatures`, even though i remember associated fns still using some kind of check.

It turns out, associated fns use [a different check altogether](https://github.com/GuillaumeGomez/rust/blob/67e8c01506d325214c5d1fd32f9456bc4580d9a3/src/librustdoc/html/render.rs#L2575-L2580):

        // FIXME(#24111): remove when `const_fn` is stabilized
        let vis_constness = if is_nightly_build() {
            constness
        } else {
            hir::Constness::NotConst
        };

However, this doesn't stop constness from showing up [in the stable std docs](https://doc.rust-lang.org/std/sync/atomic/struct.AtomicBool.html#method.new). I'm starting to think the check here to make sure we don't print constness on stable builds is superfluous. We try to compile the crate anyway; if it builds, we're on a nightly build (or a bootstrap build), by definition. If we can't hide constness from the stable std docs, then we should just leave it in and treat it like the way we treat other unstable things in the stable docs: give people a chance to look into it and shape it.

cc @rust-lang/docs Pulling in the rest of the team here; what do y'all think about this idea?

**Comment by GuillaumeGomez** [comment]
Any news in here?

**Comment by steveklabnik** [comment]
>  I'm starting to think the check here to make sure we don't print constness on stable builds is superfluous.

:+1: from me; we also had previously printed `const` ness on stable builds.

**Comment by QuietMisdreavus** [comment]
We talked about this on IRC, and are generally in favor of printing constness all the time. To this end, i'd like it if you took out the check [for bare functions, right here](https://github.com/GuillaumeGomez/rust/blob/67e8c01506d325214c5d1fd32f9456bc4580d9a3/src/librustdoc/html/render.rs#L2576) so it doesn't even try to mask it (this is the check that doesn't work and always prints it).

**Comment by QuietMisdreavus** [comment]
travis failure:

    [00:32:53] error: unused import: `rustc::session::config::nightly_options::is_nightly_build`
    [00:32:53]   --> /checkout/src/librustdoc/html/render.rs:62:5
    [00:32:53]    |
    [00:32:53] 62 | use rustc::session::config::nightly_options::is_nightly_build;
    [00:32:53]    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    [00:32:53]    |

**Comment by GuillaumeGomez** [comment]
What failure? :p

File: src/test/rustdoc/const-fn.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+#![feature(const_fn)]
+#![crate_name = "foo"]
+
+// @has foo/fn.bar.html
+// @has - '//*[@class="rust fn"]' 'pub const fn bar() -> '
+/// foo
+pub const fn bar() -> usize {
+    2
+}
```
Comment (QuietMisdreavus) [inline src/test/rustdoc/const-fn.rs]: Would you mind adding something to this test to make sure we're rendering "const" on associated functions too? Something like this is what i have in mind:

```rust
// @has foo/struct.Foo.html
// @has - '//*[@class="method"]' 'pub const fn new()'
pub struct Foo(usize);

impl Foo {
    pub const fn new() -> Foo { Foo(0) }
}
```

Comment (GuillaumeGomez) [reply src/test/rustdoc/const-fn.rs]: Sure.

**Comment by QuietMisdreavus** [comment]
r=me pending travis

**Pull request merged**